lib/liquid/c.rb

```diff
@@ -211,6 +211,19 @@ def c_find_variable_kwarg(key, raise_on_not_found: true)
   end
 end
 
+Liquid::Template.const_set("MAX_SOURCE_CODE_BYTES", (2**24) - 1)
+Liquid::Template.class_eval do
+  alias_method :ruby_parse, :parse
+
+  def c_parse(source, options = {})
+    if source.bytesize > Liquid::Template::MAX_SOURCE_CODE_BYTES
```

Comment:
I think it would be simpler do this is on an existing parsing code path, rather than introducing another monkey patch.  Also, we generally do the safety checks for C code in the C code itself, which avoids the ability for ruby code to trigger undefined behaviour.  So I would do it in `tokenizer_initialize_method` after the existing `source` checks.